Orleans | Add Azure Cosmos DB to Orleans quickstart (#37362)
dotnet/docs: docs/orleans/grains/grain-persistence/azure-cosmos-db.md

## Install NuGet package

Install the [Microsoft.Orleans.Persistence.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Persistence.Cosmos) and [Microsoft.Orleans.Clustering.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Clustering.Cosmos) NuGet packages. The Azure Cosmos DB provider stores state in a container item.

> [!IMPORTANT]
> The default database name used by the provider is **Orleans**. The default clustering container name is **OrleansCluster** and the default storage container name is **OrleansStorage**. The cluster container expects a partition key value of `/ClusterId` and the storage container expects `/PartitionKey`.

## Configure clustering provider

To configure the clustering provider use the `HostingExtensions.UseCosmosClustering` extension method.

```csharp
siloBuilder.UseCosmosClustering(
    configureOptions: static options =>
    {
        options.DatabaseName = "Orleans";
        options.DatabaseThroughput = 400;
        options.ContainerName = "OrleansCluster";
        options.ConfigureCosmosClient("AccountName=example;Key=example;");
    });
```

> ## Configure storage provider

Reviewer: ```suggestion

## Configure storage provider
```

---
title: Azure Cosmos DB grain persistence
description: Use the Azure Cosmos DB grain to persist Azure Cosmos DB for NoSQL data in a .NET Orleans application.

> author: alexwolfmsft
> ms.author: alexwolf

Reviewer: ```suggestion
```

ms.topic: conceptual
ms.devlang: csharp
ms.date: 10/03/2023
---

# Azure Cosmos DB for NoSQL grain persistence

> The Azure Cosmos DB grain persistence provider supports the [API for NoSQL](/azure/cosmos-db/nosql).

Reviewer: ```suggestion
The Azure Cosmos DB grain persistence provider supports the [NoSQL API](/azure/cosmos-db/nosql).
```

Author: We have new branding guidelines for Azure Cosmos DB that are very strict about the name. We are no longer allowed to say things like "NoSQL API," "Table API," or "MongoDB API."

It's now:

- On first mention, **Azure Cosmos DB for NoSQL**
- On any further mentions, we can say the whole thing again or shorten to **API for NoSQL**

## Install NuGet package

> Install the [Microsoft.Orleans.Persistence.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Persistence.Cosmos) and [Microsoft.Orleans.Clustering.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Clustering.Cosmos) packages from NuGet. The Azure Cosmos DB provider stores state in a container item.

Reviewer: ```suggestion
Install the [Microsoft.Orleans.Persistence.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Persistence.Cosmos) and [Microsoft.Orleans.Clustering.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Clustering.Cosmos) NuGet packages. The Azure Cosmos DB provider stores state in a container item.
```

> [!IMPORTANT]
> The default database name used by the provider is **Orleans**. The default clustering container name is **OrleansCluster** and the default storage container name is **OrleansStorage**. The cluster container expects a partition key value of `/ClusterId` and the storage container expects `/PartitionKey`.

## Configure clustering provider

> Configure the clustering provider using the `HostingExtensions.UseCosmosClustering` extension method.

Reviewer: ```suggestion
To configure the clustering provider use the `HostingExtensions.UseCosmosClustering` extension method.
```

```csharp
siloBuilder.UseCosmosClustering(

>     configureOptions: options =>
>     {
>         o.DatabaseName = "Orleans";
>         o.DatabaseThroughput = 400;
>         o.ContainerName = "OrleansCluster";
>         options.ConfigureCosmosClient("AccountName=example;Key=example;");
>     });

Reviewer: ```suggestion
    configureOptions: static options =>
    {
        options.DatabaseName = "Orleans";
        options.DatabaseThroughput = 400;
        options.ContainerName = "OrleansCluster";
        options.ConfigureCosmosClient("AccountName=example;Key=example;");
    });
```

The `o` variable isn't within scope, should be `options`.

```
## Configure storage provider

Configure the Azure Cosmos DB grain persistence provider using the `HostingExtensions.AddCosmosGrainStorage` extension method.

```csharp

> siloBuilder.AddCosmosGrainStorage(
>     name: "profileStore",
>     configureOptions: options =>
>     {
>         o.DatabaseName = "Orleans";
>         o.DatabaseThroughput = 400;
>         o.ContainerName = "OrleansStorage";
>         options.ConfigureCosmosClient("AccountName=example;Key=example;");
>     });

Reviewer: ```suggestion
siloBuilder.AddCosmosGrainStorage(
    name: "profileStore",
    configureOptions: static options =>
    {
        options.DatabaseName = "Orleans";
        options.DatabaseThroughput = 400;
        options.ContainerName = "OrleansStorage";
        options.ConfigureCosmosClient("AccountName=example;Key=example;");
    });
```
